data.py

```diff
@@ -28,17 +28,37 @@ def retrieve_paginated_data(self, suffix=None):
                       f"Content: {page_response.text}\nData processed before the error returned")
         return data
 
-    def get_data(self, rec_id=None, output='df'):
+    def get_data(self, rec_id=None, name=None, unique_key_available=None, refresh=None, output='df'):
         """Retrieve list of data sources or retrieve the contents of a data source
         rec_id : int
             RecId of the data you want to retrieve
+        name : str
+            data set name based filter to apply
+        unique_key_available : bool
+            filter data sets based on whether they're using unique keys
+        refresh : bool
+            filter data sets based on whether they're refresh-able (non-static)
         output : str
             Output format, either df or dict
         """
         if rec_id is None:
-            # Check token for expiry
-            data = self.session.api_call(suffix='data', request_method='get')
-            # return data.json()
+            # Add parameter string based on input
+            params = {}
+            if type(name) == str:
+                params["name"] = name
+            if type(unique_key_available) == bool:
+                params["uniquekeyavailable"] = unique_key_available
+            if type(refresh) == bool:
+                params["refresh"] = refresh

```

Comment:
There is a style of programming, specially with dynamic typed languages like Python, where developers tend to check for a variable type closer to where the variable is used. The reason being is that the type can change at any time, therefore an early check is just wasteful. If you are using Python 3.5 or later, you may want to use [Type Annotations](https://dev.to/dstarner/using-pythons-type-annotations-4cfe) instead to give a hint of what the type should be. Regardless, I would just simple not check for the type until I am ready to use the values in the variables.